### DISCUSSION TITLE
Glitches after disabling modern stand by. Doc for settings?

### DISCUSSION THREAD

**Discussion by cristian-spiescu** [DavidS95/Smokeless_UMAF#10] [Q&A]
Hi,

I own a Legion 5 Pro AMD 6800H + RTX3060. I used this tool to disable modern stand by (s0) in favor of the classic stand by (s3). This seems to have introduced a side effect: I have issues during screen share because of graphics card driver crash/restore. Details [here](https://forums.lenovo.com/t5/Gaming-Laptops/Legion-5-Pro-AMD-6800H-RTX3600-graphics-driver-crashing-during-screen-share/m-p/5205459).

I see some settings that might be related to the graphics card. But I'm wondering if I could find somewhere detailed explanation of the settings?

Not only because trying them one by one is a bit time consuming, but I'm also afraid of bricking my unit.

Thanks.

**Comment by DavidS95** [comment]
I dont know much about how disabling modern standby would cause issues for the graphics card. How sure are you that its regarding this s3 setting? Other people I know have used this setting with a nvidia dgpu prior and not experienced this issue. Idk if this issue is exclusive to your laptop model (or similar from lenovo) or not. But theres no guarentee classic s3 will work, i know others where it didnt work outright. But what i do know is that, there is nothing on the programs side that can be done about this issue. Id recommend reinstalling the gpu driver if u havent already, maybe it will detect u have no modern standby and work differently accordingly. Also try a different power profile in windows, or in nvidia control pannel that might change how the gpu behaves.

**Comment by cristian-spiescu** [comment]
Thanks for your answer.

> How sure are you that its regarding this s3 setting?

W/ S0, the issue doesn't appear.
W/ S3, after boot, the issue doesn't appear. If I put the PC in sleep + wake up: then the issue appears. So the GPU is somehow in some kind of weird state after wake up.

> Other people I know have used this setting with a nvidia dgpu prior and not experienced this issue.

I emphasize on the fact that the issue is really particular: crash ONLY when screen sharing in some programs (not all). E.g. skype, teams. So they are probably using a particular API call. All other things (about the GPU) seem to function correctly. So maybe there are people w/ this issue, but because they don't use this scenario, they are not affected.

> is exclusive to your laptop model (or similar from lenovo) 

My intuition tells me that it is specific to this model. The previous laptop, Legion 5, 5800H, RTX 3060 didn't have this issue. But neither had it S0 enabled. So it came w/ S3 configured by the factory.

> nothing on the programs side that can be done

I agree.

My current workaround is a `.bat` script that I run before doing screen share.

```
pnputil /disable-device "PCI\VEN_10DE&DEV_2560&SUBSYS_3B2B17AA&REV_A1\4&1756E899&0&0009"
pnputil /enable-device "PCI\VEN_10DE&DEV_2560&SUBSYS_3B2B17AA&REV_A1\4&1756E899&0&0009"
```

This is the equivalent of the manual actions: disable/enable the dGPU in Device Manager.

I have played a bit w/ some parameters I saw in the program. W/o luck. Also: via google search, I didn't find additional doc for them. Do you know if such doc exists?

**Comment by rp1231** [comment]
I have the same laptop Lenovo Legion 5 pro but with 6800 + 3070 TI  and I had used it for the same thing(enabling the classic s3 standby).
It worked fine for me except for troubles with the wifi not activating after waking up from sleep.

I thought that there were no problems apart from this but after a certain amount of time (Can't accurately recall how much) my laptop got bricked.
Not sure if it happened due to the tinkering with the sleep states or if it was a hardware failure, 
but posting it here as I had the same specs.
Thankfully my laptop was new and under warranty so I've currently sent it for repairs.

**Reply by grotesque** [reply]
_it happened due to the tinkering with the sleep states or if it was a hardware failure_
@rp1231 

That's scary. Can you elaborate more?

What do you mean it got bricked? Were you unable to load windows or were you unable to even get to the uefi-bios? What did you do just before it got bricked? Anything more will help. 

Thanks!

**Reply by rp1231** [reply]
I had just put it to sleep or shut it down iirc.
But when I started it again, it never woke up.
Only the keyboard rgb lights would turn on and nothing else.
The fans would spin for a few seconds and then stop.
I wasn't able to get to the uefi-bios or use the form browser usb.
My guess is that it was probably a hardware failure as other people have encountered the same problem on reddit.
But just to be on the safe side I don't wan't  to mess with the setting again.

**Comment by cristian-spiescu** [comment]
As a curiosity: did you/do you do screen share from skype or teams?

> worked fine for me except for troubles with the wifi
In my case, I had an issue w/ the keyboard. At that time, I didn't associate it w/ S3. The solution was to disable selective suspend (in power options). But now, I'm thinking that this S3 thing is the one to blame.

I also had issus w/ Bluetooth. Doing a "reassociate" on a particular device, would consistently give me blue screens. After unpair + pair, no more issues.

And besides this, I observed other small glitches. I'm under the impression that Windows and/or the BIOS have somehow a self-healing mechanism. And (only in specific cases), if they detect an issue, then they toggle some flags, and the issue doesn't appear any more.

All what I'm writing here did not exist w/ my previous laptop (Legion 5, 5800H, RTX3060). So I guess that toggling S3 "officially", needs additional steps, that we don't know. I'm under the impression that some Thinkpads have this toggle as an official BIOS option.

**Comment by rp1231** [comment]
Nope, didn't use it for screen share ever.....
All I know is that I'm not going to be messing with any of these settings if/when my laptop comes back from the service center.
Not worth it for me at least...

**Comment by Gamrix** [comment]
I found that with my Lenovo Thinkbook 13 gen 3s, (almost same Ryzen 6800u chip), that after some updates, the device soft bricked itself. (I was able to load the BIOS, but Windows would blue screen with an error saying it couldn't find the boot device. Most of the settings to undo this would not work, but I was able to get it to unbrick by booting windows up in Safe Mode. During the startup for this, it was able to revert the update, but in doing so also reverted the s3 sleep BIOS changes (I don't know what other bios changes it did).

**Comment by cristian-spiescu** [comment]
I had maybe a similar issue after a BIOS UPDATE. Windows didn't boot any more w/ a dubious error message (that was not helpful for searching solutions on the Internet). Solution: boot in safe mode. Attempting a repair via Win RE didn't work. 

Lesson learned: activate the "dual bios/keep last bios version".

During my trials I "revert to default" the BIOS. By doing this, I lost my S3 settings. So after this, I needed to open Smokeless UMAF again, to redo the S3 setting.

I mention that I used this opportunity to give a new chance to Modern standby. Unfortunately it disappointed me again, because of the high battery drain during stand by. And looked again w/ wishing eyes towards the integrated platform hardware + software of Mac.

**Comment by MDBelov** [comment]
On Legion 7 AMD 6800h/6900hx + 6850M XT models, to solve problems with switching to s3, we came to the following settings:
Device Manager > AMD PBS > Power saving configuration
S3/Modern Standby Support - S3 Enabled
Wake on PME - Enabled
Internal PCIe GPP 0 D3 - Enabled

**Reply by cristian-spiescu** [reply]
Great contribution! I also started myself a trial/error session to find a "magic combo". However w/o success.

BTW: when you say "problems": what exact problems did you have?

**Reply by MDBelov** [reply]
The laptop did not wake up from the bluetooth mouse and over the network. Wi-Fi had to be turned on manually each time.

**Reply by cristian-spiescu** [reply]
I see; so you we are speaking than different problems, compared w/ mine. In my case I had issues w/ the integrated display adapter, during screen share in SOME programs. However, I notice that since a few weeks they seem to have gone. Maybe after 1000 years a proper driver update came via Windows updates.

> The laptop did not wake up from the bluetooth mouse.

I didn't know that this is possible actually. Maybe the `Wake on PME` settings is responsible for this, judging by its name from your list.

**Reply by MDBelov** [reply]
It seems that the problems are individual for each laptop model. Above there were complaints about problems with Wifi on another Legion.

**Comment by Espionage724** [comment]
There's a S3 video repost option somewhere under ACPI options; I never noticed it do anything but it sounds like it might be useful here?
